.github/workflows/test-python.yaml

```diff
@@ -5,25 +5,52 @@ on:
   push:
     branches: [main]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
IIRC, we discussed before that this part is not needed given that we don't have limitations on concurrent runs due to GitHub enterprise.

Reply:
A use case is re-pushing the branch for the PR while the workflow is running. Even with no limitations, is that useful to keep the existing run going?

Reply:
Might be useful to compare results, but I am fine to cancel them for now.
@kramaranya Please can you apply those changes to all GitHub action tests in that case ?